workflow/controller/workflowpod.go

```diff
@@ -414,17 +416,22 @@ func (woc *wfOperationCtx) createWorkflowPod(ctx context.Context, nodeName strin
 
 	woc.log.Debugf("Creating Pod: %s (%s)", nodeName, nodeID)
 
-	created, err := woc.controller.kubeclientset.CoreV1().Pods(woc.wf.ObjectMeta.Namespace).Create(ctx, pod, metav1.CreateOptions{})
-	if err != nil {
-		if apierr.IsAlreadyExists(err) {
-			// workflow pod names are deterministic. We can get here if the
-			// controller fails to persist the workflow after creating the pod.
-			woc.log.Infof("Failed pod %s (%s) creation: already exists", nodeName, nodeID)
-			return created, nil
-		}
-		if errorsutil.IsTransientErr(err) {
```

Comment:
It should already be retried in transient errors. The controller would requeue this when transient error happens.

Reply:
Ref: https://github.com/argoproj/argo-workflows/blob/master/workflow/controller/operator.go#L2234-L2242

Reply:
Can you provide what kind of error it is failing? if it is in transient error which missed in requeueIfTransientErr, you can add it

Reply:
We've observed two errors (both of type *url.Error) with the following messages:
Post https://192.168.254.1:443/api/v1/namespaces/delivery/pods: read tcp 10.244.56.73:35718->192.168.254.1:443: read: connection reset by peer"

Reply:
It looks like your k8s API server is unavailable or high load. But in normal scenarios, these errors are not considered transient errors. I would recommend adding retrystrategy on those steps.  `Backoff` will create an extra load on API server.